File: for-loops/for.js

var i = 5;

for (var i=0; i<2; i++) {
    console.log("for:"+i); // prints 0, 1
}

console.log("after:"+i); // prints 2

i = 5;

for (var j=0; j<2; j++) {
    var i = j+1;
    console.log("for2:"+i); // prints 1, 2
}

console.log("after:"+i); // prints 2

File: for-loops/for.pl

$i = 5;

for (local $i=0; $i<2; $i++) {
  print "for:$i\n"; # prints 0, 1
}

print "after:$i\n"; # prints 2

$i = 5;

for ($j=0; $j<2; $j++) {
  local $i = $j+1;
  print "for2:$i\n"; # prints 1, 2
}

print "after:$i\n"; # prints 5
